platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -0,0 +1,1036 @@
+  shares.
+- Conversely, if you want to collect more than 2/3 the weight while holding as few shares as possible, that happens when
+  you collect only shares with wps of 1, and all the rest are 1/2. And again, that means you have more than 1/2 the
+  shares.
+- So both theorems are proved.
+
+###### Threshold for Votes to Adopt a Candidate Roster
+
+The threshold for votes to adopt a candidate roster will be at least 1/3 of the consensus weight to ensure that at
+least 1 honest node has validated the TSS key material.
+
+###### Impacts on Service's Calculation of Weight from Staked HBars
+
+The services layer will track staking to nodes, clipping any stake above the max, and reducing any stake below `1/R` of
+the maximum clipped stake to zero (where `R` is a setting of services, set to twice the max number of shares allowed per
+node).
+
+The resulting stake (after clipping or zeroing) is passed to the platform layer, which treats it as a “weight” for
+consensus, and which then generates a number of shares for each node. That share count is proportional to that weight,
+and acts as a low-resolution version of that weight.
+
+##### TSS Algorithm - Alternatives Considered
+
+The list of options considered here were based off of prototypes developed by [Rohit Sinha](https://github.com/rsinha),
+the Hashgraph cryptography expert. The Groth21 algorithm was chosen because it is efficient for use in smart
+contract verification, and we can assign a multiplicity of shares to nodes to get close enough in modeling the
+distribution of weight between nodes.
+
+| Requirement | hinTS | Groth21 |
+|-------------|-------|---------|
+| TSS-001     | Yes   | Yes     |
+| TSS-002     | Yes   | Yes     |
+| TSS-003     | Yes   | Yes     |
+| TSS-004     | Yes   | Yes     |
+| TSS-005     | Yes   | Yes     |
+| TSS-006     | Yes   | Yes     |
+| TSS-007     | Yes   | Yes     |
+| TSS-008     | No    | Yes     |
+| TSS-009     | Yes   | Yes     |
+| TSS-010     | Yes   | No      |
+
+###### hinTS
+
+The hinTS algorithm is a threshold signature scheme that is able to measure double precision distributions of weight
+across nodes. The complicating factor with hinTS is that during an address book change over, a recursive proof needs
+to be constructed to prove that the new roster is a descendent of the original genesis roster. Validation of this
+recursive proof proved too expensive for EVM smart contracts, even with elliptic curve precompiles.
+
+#### TSS Bootstrap Process
+
+The TSS bootstrap process is the process of generating the ledger public/private key pair for an
+existing network. There are several options to consider in the TSS bootstrap process:
+
+1. Do we allow block number and round number to be out of sync?
+2. Do we need the initial block produced to be signed by the ledger id?
+3. For new networks, do we need to produce a signed block 1?
+4. For existing networks, does the TSS bootstrap roster sign blocks prior to its existence?
```

Comment:
Are the answers to these questions depending on block steams?

Reply:
Yes.  I am updating this proposal with answers that have been determined so far.